chips/stm32f4xx/src/fsmc.rs

```diff
@@ -0,0 +1,358 @@
+        WAITEN OFFSET(13) NUMBITS(1) [],
+        /// Write enable bit
+        WREN OFFSET(12) NUMBITS(1) [],
+        /// Wait timing configuration
+        WAITCFG OFFSET(11) NUMBITS(1) [],
+        /// Wait signal polarity bit
+        WAITPOL OFFSET(9) NUMBITS(1) [],
+        /// Burst enable bit
+        BURSTEN OFFSET(8) NUMBITS(1) [],
+        /// Flash access enable
+        /// Enables NOR Flash memory access operations
+        FACCEN OFFSET(6) NUMBITS(1) [],
+        /// Memory data bus_width width
+        MWID OFFSET(4) NUMBITS(2) [
+            BITS_8 = 0b00,
+            BITS_16 = 0b01
+        ],
+        /// Memory type
+        MTYP OFFSET(2) NUMBITS(2) [
+            SRAM = 0b00,
+            PSRAM = 0b01,
+            NOR = 0b10
+        ],
+        /// Address/data multiplexing enable bit
+        MUXEN OFFSET(1) NUMBITS(1) [],
+        /// Memory bank enable bit
+        MBKEN OFFSET(0) NUMBITS(1) []
+    ],
+    BTR [
+        /// Access mode
+        ACCMOD OFFSET(28) NUMBITS(2) [
+            A = 0b00,
+            B = 0b01,
+            C = 0b10,
+            D = 0b11
+        ],
+        /// Data latency for synchronous memory
+        DATLAT OFFSET(24) NUMBITS(4) [],
+        /// Clock divide ratio (for FSMC_CLK signal)
+        CLKDIV OFFSET(20) NUMBITS(4) [],
+        /// Bus turnaround phase duration
+        BUSTURN OFFSET(16) NUMBITS(4) [],
+        /// Data-phase duration
+        DATAST OFFSET(8) NUMBITS(8) [],
+        /// Address-hold phase duration
+        ADDHLD OFFSET(4) NUMBITS(4) [],
+        /// Address setup phase duration
+        ADDSET OFFSET(0) NUMBITS(4) []
+    ],
+    BWTR [
+        /// Access mode
+        ACCMOD OFFSET(28) NUMBITS(2) [
+            A = 0b00,
+            B = 0b01,
+            C = 0b10,
+            D = 0b11
+        ],
+        BUSTURN OFFSET(16) NUMBITS(4) [],
+        /// Data-phase duration
+        DATAST OFFSET(8) NUMBITS(8) [],
+        /// Address-hold phase duration
+        ADDHLD OFFSET(4) NUMBITS(4) [],
+        /// Address setup phase duration
+        ADDSET OFFSET(0) NUMBITS(4) []
+    ]
+];
+
+/// This mechanism allows us to schedule "interrupts" even if the hardware
+/// does not support them.
+static DEFERRED_CALL: DeferredCall<DeferredCallTask> =
+    unsafe { DeferredCall::new(DeferredCallTask::Fsmc) };
+
+const FSMC_BASE: StaticRef<FsmcBankRegisters> =
+    unsafe { StaticRef::new(0xA000_0000 as *const FsmcBankRegisters) };
+
+/// FSMC Bank
+#[repr(C)]
+struct FsmcBank {
+    /// Address
+    reg: ReadWrite<u16>,
+    /// Data
+    ram: ReadWrite<u16>,
+}
+
+const FSMC_BANK1: StaticRef<FsmcBank> = unsafe { StaticRef::new(0x60000000 as *const FsmcBank) };
+const FSMC_BANK2_RESERVED: StaticRef<FsmcBank> = unsafe { StaticRef::new(0x0 as *const FsmcBank) };
+const FSMC_BANK3: StaticRef<FsmcBank> = unsafe { StaticRef::new(0x68000000 as *const FsmcBank) };
+const FSMC_BANK4_RESERVED: StaticRef<FsmcBank> = unsafe { StaticRef::new(0x0 as *const FsmcBank) };
```

Comment:
I don't like this practice of treating an arbitrary memory location as though it is an `FsmcBank`. This could easily lead to memory unsafety if these slots were used. If empty slots are needed in the FSMC array, the array should hold `Option<StaticRef<FsmcBank>>`

Reply:
Thanks for the tip, I modified it.